### PR TITLE
Optional gen-run style wrapping in awaitify.module

### PR DESCRIPTION
**Pull request by mrlotfi**
A new parameter is added to awaitify.module as an option to avoid wrapping function generators in gen-run style callbacks.

### CODE REVIEW

File: lib/module.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function (baseModule, moduleDefinition) {
     .keys()
     .each((attributeKey) => {
       const value = moduleDefinition[attributeKey];
-      moduleObject[attributeKey] = isGenerator(value) ? wrapInGenRunCallback(value.awaitify()) : value;
+      moduleObject[attributeKey] = (isGenerator(value) && !doNotWrapInGenRunCallback) ? wrapInGenRunCallback(value.awaitify()) : value;
```
Comment (mehranhatami) [inline lib/module.js]: @mrlotfi you should consider the case when the value is a generator and doNotWrapInGenRunCallback is true. The correct implementation is like this:
```javascript
moduleObject[attributeKey] = isGenerator(value) ? doNotWrapInGenRunCallback ?  value.awaitify() : wrapInGenRunCallback(value.awaitify()) : value;
```

File: lib/module.js [hunk 2]
```diff
@@ -3,8 +3,8 @@ require('./core');
 const isGenerator = require('./isGenerator');
 const defaultBaseModule = {};
 
-module.exports = function (baseModule, moduleDefinition) {
-  if (arguments.length === 1) {
+module.exports = function (baseModule, moduleDefinition, doNotWrapInGenRunCallback) {
+  if (arguments.length === 1 || moduleDefinition === undefined) {
```
Comment (mehranhatami) [inline lib/module.js]: @mrlotfi you could rewrite this like this:
```javascript
module.exports = function (baseModule, moduleDefinition, doNotWrapInGenRunCallback) {
  if (arguments.length === 1) {
    moduleDefinition = baseModule;
    baseModule = defaultBaseModule;
    doNotWrapInGenRunCallback = false;
  } else if (arguments.length === 2) {
    if(typeof moduleDefinition === 'boolean'){
      moduleDefinition = baseModule;
      baseModule = defaultBaseModule;
      doNotWrapInGenRunCallback = moduleDefinition;
    } else {
      doNotWrapInGenRunCallback = false;
    }
  }
  // THE REST OF THE CODE
};
```

File: lib/module.js [hunk 3]
```diff
@@ -3,8 +3,8 @@ require('./core');
 const isGenerator = require('./isGenerator');
 const defaultBaseModule = {};
 
-module.exports = function (baseModule, moduleDefinition) {
-  if (arguments.length === 1) {
+module.exports = function (baseModule, moduleDefinition, doNotWrapInGenRunCallback) {
```
Comment (mehranhatami) [inline lib/module.js]: @mrlotfi I reviewed everything, consider making this changes. Would you please write some tests for the third argument.

**Review by mehranhatami — COMMENTED** [review]
@mrlotfi please don't forget to write some tests or edit the tests.

**Comment by mrlotfi** [comment]
I've considered your review issues. Would you please re-review?

**Review by mehranhatami — APPROVED** [review]
Well done!

**Pull request merged**